propagation/b3/http.go

```diff
@@ -21,52 +21,104 @@ import (
 	"github.com/openzipkin/zipkin-go/propagation"
 )
 
+type InjectOption func(opts *InjectOptions)
+
+type InjectOptions struct {
+	shouldInjectSingle   bool
+	shouldInjectMultiple bool
+}
+
+// WithInjectSingleAndMultiple allows to include both single and multiple
+// headers in the context injection
+func WithInjectSingleAndMultiple() InjectOption {
+	return func(opts *InjectOptions) {
+		opts.shouldInjectSingle = true
+		opts.shouldInjectMultiple = true
+	}
+}
+
+// WithInjectSingleOnly allows to include only single header in the context
+// injection
+func WithInjectSingleOnly() InjectOption {
+	return func(opts *InjectOptions) {
+		opts.shouldInjectSingle = true
+		opts.shouldInjectMultiple = false
+	}
+}
+
 // ExtractHTTP will extract a span.Context from the HTTP Request if found in
 // B3 header format.
 func ExtractHTTP(r *http.Request) propagation.Extractor {
-	return func() (*model.SpanContext, error) {
+	return func() (sc *model.SpanContext, err error) {
```

Comment:
I think I'd remove the return variable declarations to be more explicit about returns...